### PR TITLE
[WIP] Kill whole group, not just parent

### LINKED ISSUE #50: Stop command should kill whole process group
If the top-level (daemonized) service starts one or more long-running child processes, it has to ensure that those child processes get killed when the parent process is killed. This is somewhat painful to do in Java because it involves setting a shutdown hook, and it's repetitive work that's easy to get wrong. If done incorrectly, the child processes become zombies and have to be reaped manually by other means; they could take up system resources such as memory and ports.

Instead, if `init.sh` were to record the process group ID and `kill` the process group, we'd have a clean solution that works without additional work from the app.

Tagging @mingyukim, @tomkozlowski and @gdearment for comments.

### PR DESCRIPTION
**Pull request by punya**
Fixes #50.

### CODE REVIEW

File: src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy [hunk 1]
```diff
@@ -279,6 +279,38 @@ class JavaDistributionPluginTests extends Specification {
...
+        buildResult.task(':distTar').outcome == TaskOutcome.SUCCESS
+        buildResult.task(':untar').outcome == TaskOutcome.SUCCESS
+
+        // check content was extracted
+        new File(projectDir, 'dist/service-name-0.1').exists()
+
+        // try all of the service commands
+        String output = exec('dist/service-name-0.1/service/bin/init.sh', 'start')
+        exec('dist/service-name-0.1/service/bin/init.sh', 'stop')
+        exec("ps aux | grep '[y]es'").trim() == ""
```
Comment (markelliot) [inline src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy:311]: the `[y]` here is to distinguish the grep command from finding itself?

Comment (punya) [inline src/test/groovy/com/palantir/gradle/javadist/JavaDistributionPluginTests.groovy:311]: Yeah, I found this trick on a Stackoverflow page.

**Comment by markelliot** [comment]
In principle I'm fine with this, but the tests are failing =/

**Comment by markelliot** [comment]
@punya still interested in pushing this through?

**Comment by punya** [comment]
Yes but I won't be able to work on it for a few more days.

**Comment by markelliot** [comment]
@punya I'm going to close this for now, feel free to reopen if you'd like to contribute the change to init.sh. @uschi2000 is working on a go-based launcher to clean up #73 and I think we can account for process group as part of that.

**Pull request closed without merging**